pkg/kernel/nfqueue.go

```diff
@@ -131,139 +152,315 @@ func (nfq *NFQueue) Update(protocols []string, sourceIPs []string, destinationIP
 func (nfq *NFQueue) Delete() error {
 	nfq.mu.Lock()
 	defer nfq.mu.Unlock()
-	var errFinal error
-	err := execCommand(fmt.Sprintf("nft delete chain inet %s %s", tableName, nfq.name))
-	if err != nil {
-		errFinal = fmt.Errorf("%w; %v", errFinal, err)
-	}
-	err = execCommand(fmt.Sprintf("nft delete set inet %s %s", tableName, nfq.getFullSetName(ipv4DestinationIPSet)))
-	if err != nil {
-		errFinal = fmt.Errorf("%w; %v", errFinal, err)
+	conn := &nftables.Conn{}
+	conn.DelChain(nfq.chain)
```

Comment:
What about the rules? Was Flow removal tested?
https://wiki.nftables.org/wiki-nftables/index.php/Configuring_chains#Deleting_chains
"The only condition is that the chain you want to delete needs to be empty, otherwise the kernel will complain that the chain is still in use."

Reply:
Right, I tried it without removing the rules, and it worked fine, traffic was no longer working, and I could create the same flow again after removing it

Reply:
That would mean, that the documentation is misleading.
Could you please double check whether during Flow removal an error is received in the code upon deleting the chain? And check how the nftables looks like after a Flow removal?

Reply:
I will check again if there any error. Otherwise I tried this morning again, and I checked the nftables, everything that belongs to the flow was removed correctly (no more set, no more rule, no more chain). And I triied to create it again, everything was fine (no duplicates or anything)

Reply:
I just tried, and no, I don't see any error on flow removal